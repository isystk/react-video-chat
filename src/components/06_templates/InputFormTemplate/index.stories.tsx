import { Meta, Story } from '@storybook/react'
import React from 'react'
import InputFormTemplate from './index'
import MainService from '@/services/main'

export default {
  title: '06_templates/InputFormTemplate',
  component: InputFormTemplate,
} as Meta

const Template: Story = ({ name = '' }) => {
  const main = new MainService(() => ({}))
  main.setName(name)
  const props: InputFormTemplate = {main}
  return (
    <InputFormTemplate {...props} />
  )
}

export const InputYourName = Template.bind({})
InputYourName.storyName = 'あなたのお名前を入力'
InputYourName.args = {}

export const InputRoomName = Template.bind({})
InputRoomName.storyName = '部屋の名前を入力'
InputRoomName.args = { name: 'isystk' }
