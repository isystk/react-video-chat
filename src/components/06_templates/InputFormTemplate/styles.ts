import { css, keyframes } from '@emotion/css'

export const inputTemplate = css``
