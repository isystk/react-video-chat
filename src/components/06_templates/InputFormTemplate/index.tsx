import Grid from '@material-ui/core/Grid'
import Circles from '@/components/01_atoms/Circles'
import InputFormName from '@/components/03_molecules/InputFormName'
import InputFormRoom from '@/components/03_molecules/InputFormRoom'
import React, { useEffect, FC, useContext } from 'react'
import { useRouter } from 'next/router'
import DeviceSettingModal from '@/components/04_organisms/DeviceSettingModal'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'

/** InputFormTemplateProps Props */
export type InputFormTemplateProps = WithChildren
/** Presenter Props */
export type PresenterProps = InputFormTemplateProps & { classes }

/** Presenter Component */
const InputFormTemplatePresenter: FC<PresenterProps> = ({
  main,
  classes,
  ...props
}) => (
  <>
    <div className="area">
      <Circles>
        <Grid container spacing={0}>
          <Grid item xs={12}>
            <InputFormName />
            <InputFormRoom />
          </Grid>
        </Grid>
        <DeviceSettingModal />
      </Circles>
    </div>
  </>
)

/** Container Component */
const InputFormTemplateContainer: React.FC<
  ContainerProps<InputFormTemplateProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const classes = useStyles()
  const main = useContext(Context)
  const router = useRouter()

  useEffect(() => {
    if (main.self.name && main.room.roomId) {
      router.push(main.room.roomId)
    }
  }, [main.self.name, main.room.roomId])
  return presenter({ children, main, classes, ...props })
}

export default connect<InputFormTemplateProps, PresenterProps>(
  'InputFormTemplate',
  InputFormTemplatePresenter,
  InputFormTemplateContainer
)
