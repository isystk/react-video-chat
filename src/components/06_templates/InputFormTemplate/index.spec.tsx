import React from 'react'
import renderer from 'react-test-renderer'
import InputFormTemplate, {InputFormTemplateProps} from './index'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'
import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import { Provider } from 'react-redux'
import reducers from '@/stores'

describe('InputFormTemplate', () => {
  it('Match Snapshot', () => {
    const enhancer = applyMiddleware(thunk)
    const store = createStore(reducers, enhancer)
    const main = new MainService(() => ({}))
    const props: InputFormTemplateProps = {main}
    const component = renderer.create(
      <Provider store={store}>
        <InputFormTemplate {...props} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
