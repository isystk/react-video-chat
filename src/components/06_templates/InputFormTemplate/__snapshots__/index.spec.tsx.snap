// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFormTemplate Match Snapshot 1`] = `null`;
